core/src/main/java/com/netflix/iceberg/SnapshotUpdate.java

```diff
@@ -273,4 +328,31 @@ private static ManifestFile addMetadata(TableOperations ops, ManifestFile manife
       throw new RuntimeIOException(e, "Failed to read manifest: %s", manifest.path());
     }
   }
+
+  private static void updateTotal(ImmutableMap.Builder<String, String> summaryBuilder,
+                                  Map<String, String> previousSummary, String totalProperty,
+                                  Map<String, String> currentSummary,
+                                  String addedProperty, String deletedProperty) {
+    String totalStr = previousSummary.get(totalProperty);
+    if (totalStr != null) {
+      try {
+        long newTotal = Long.parseLong(totalStr);
+
+        String addedStr = currentSummary.get(addedProperty);
+        if (addedStr != null) {
+          newTotal += Long.parseLong(addedStr);
+        }
+
+        String deletedStr = currentSummary.get(deletedProperty);
+        if (deletedStr != null) {
+          newTotal -= Long.parseLong(deletedStr);
+        }
+
+        summaryBuilder.put(totalProperty, String.valueOf(newTotal));
+
+      } catch (NumberFormatException e) {
+        // ignore and do not add total
```

Comment:
Would you consider logging a warning message for this exception though?

Reply:
Yes, a log message would be useful here.